runtime/src/test/scala/com/fortysevendeg/exercises/EvaluatorSpec.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * scala-exercises-runtime
+ * Copyright (C) 2015-2016 47 Degrees, LLC. <http://www.47deg.com>
+ */
+
+package org.scalaexercises.runtime
+
+import scala.concurrent.duration._
+import org.scalatest._
+
+class EvaluatorSpec extends FunSpec with Matchers {
+
+  describe("evaluation") {
+    it("fails with a timeout when takes longer than the configured timeout") {
+      val evaluator = new Evaluator(1 second)
+      val result: EvalResult[Int] = evaluator("", "{ while(true) {}; 123 }")
```

Comment:
In that case it'd be `EvalRuntimeError` containing the thrown exception, which is of type `EvalResult[T]`. The original exception is captured and its message shown to the user in the UI.